thirdparty/github.go

```diff
@@ -375,9 +391,14 @@ func GetGithubFile(ctx context.Context, token, owner, repo, path, ref string) (*
 	}
 
 	// Fallback to not using the GitHub app on error.
-	legacyClient := getGithubClientRetry(token, "GetGithubFile")
-	defer utility.PutHTTPClient(legacyClient)
-	client := github.NewClient(legacyClient)
+	ctx, httpLegacyClient, putLegacyClient := getGithubClient(ctx, token, "GetGithubFile", retryConfig{retry: true}, []attribute.KeyValue{
+		attribute.String(githubOwnerAttribute, owner),
+		attribute.String(githubRepoAttribute, repo),
+		attribute.String(githubRefAttribute, ref),
+		attribute.String(githubPathAttribute, path),
+	})
+	defer putLegacyClient()
+	client := github.NewClient(httpLegacyClient)
```

Comment:
Discussed below -- including `if newResp.StatusCode == http.StatusNotFound` up here should be good

Reply:
added to the if statement above